### PR TITLE
Create new .NET 6.0 pages

### PR DESCRIPTION
**Pull request by RussKie**
- Move .NET 5.0
- Add new .NET 6.0

[Internal Preview](https://review.docs.microsoft.com/en-us/dotnet/desktop/winforms/whats-new/net60?view=netdesktop-5.0&branch=pr-en-us-1199)

### CODE REVIEW

**Comment by adegeo** [comment]
@RussKie I had reformatted the article while you made those questions. Would you be able to port them to the latest version? I would do it but I don't want to mess something up.

File: .openpublishing.redirection.json [hunk 1]
```diff
@@ -528,6 +528,10 @@
         {
             "source_path": "dotnet-desktop-guide/framework/wpf/properties/custom-dependency-properties.md",
             "redirect_url": "/dotnet/desktop/wpf/advanced/custom-dependency-properties?view=netframeworkdesktop-4.8"
+        },
+        {
+            "source_path": "dotnet-desktop-guide/net/winforms/whats-new/index.md",
+            "redirect_url": "/dotnet/desktop/winforms/whats-new/net60?view=netdesktop-5.0"
```
Comment (RussKie) [inline .openpublishing.redirection.json:534]: This looks strange "net60?view=netdesktop-5.0"

File: redirects_generator/definitions.json [hunk 2]
```diff
@@ -235,6 +235,15 @@
             "TargetUrl": "/dotnet/desktop/winforms/input-mouse/how-to-distinguish-between-clicks-and-double-clicks?view=netdesktop-5.0"
         },
 
+    // ========
+    // WinForms Index mappings
+    // ========
+        {
+            "Redirect": "Normal",
+            "SourceUrl": "/dotnet/desktop/winforms/whats-new/index?view=netdesktop-5.0",
+            "TargetUrl": "/dotnet/desktop/winforms/whats-new/net60?view=netdesktop-5.0"
```
Comment (RussKie) [inline redirects_generator/definitions.json:244]: ditto

**Review by RussKie — COMMENTED** [review]
@adegeo why did you delete whats-new/index.md, that contained the skeleton for .NET 6.0 changeset?

File: .openpublishing.redirection.json (hunk 1 above)
Comment (adegeo) [reply .openpublishing.redirection.json:534]: I need to do a PR in the repo that just mass updates these from 5.0 to 6.0. At the time the 6.0 moniker didn't exist.

**Comment by adegeo** [comment]
@RussKie I deleted index.md and I'm using a redirect to point to the latest "what's new" so in this cases, 6.0

File: dotnet-desktop-guide/net/winforms/whats-new/index.md [hunk 3]
```diff
@@ -7,56 +7,119 @@ ms.topic: conceptual
...
+- <xref:System.Windows.Forms.LinkClickedEventArgs?displayProperty=fullName> has now a new constructor and extended with the following properties:
+    - <xref:System.Windows.Forms.LinkClickedEventArgs.LinkLength?displayProperty=fullName>
+    - <xref:System.Windows.Forms.LinkClickedEventArgs.LinkStart?displayProperty=fullName>
+
+- <xref:System.Windows.Forms.NotifyIcon.Text?displayProperty=fullName> is now limited to 127 characters (from 63).
+
+## Enhanced features
+
+- High DPI improvements.
+TODO: @dreddy-work to provide text.
```
[the remark this answers is not in the document]
Comment (dreddy-work) [reply dotnet-desktop-guide/net/winforms/whats-new/index.md]: Will get to it Today or Tomorrow.

File: dotnet-desktop-guide/net/winforms/whats-new/index.md [hunk 4]
```diff
@@ -7,56 +7,119 @@ ms.topic: conceptual
...
+## Updated templates
+
+In line with the general .NET direction we have updated our C# templates with the following features:
+- `ApplicationConfiguration.Initialize()` bootstrap API, 
+- [file-scope namespace declarations](https://devblogs.microsoft.com/dotnet/announcing-net-6-release-candidate-2/#file-scoped-namespace-declaration),
+- nullable support,
+- [implicit usings support](https://devblogs.microsoft.com/dotnet/announcing-net-6-release-candidate-2/#implicit-usings),
+- [global usings support](https://devblogs.microsoft.com/dotnet/announcing-net-6-release-candidate-2/#global-usings).
+
+TODO: note about .NET Framework templates. @OliaG to provide text.
```
[the remark this answers is not in the document]
Comment (merriemcgaw) [reply dotnet-desktop-guide/net/winforms/whats-new/index.md]: @RussKie last time I checked they were actually back as installed by default. So perhaps we don't need this particular todo.

File: dotnet-desktop-guide/net/winforms/whats-new/index.md [hunk 5]
```diff
@@ -7,56 +7,119 @@ ms.topic: conceptual
...
+    - <xref:System.Windows.Forms.LinkClickedEventArgs.LinkStart?displayProperty=fullName>
+
+- <xref:System.Windows.Forms.NotifyIcon.Text?displayProperty=fullName> is now limited to 127 characters (from 63).
+
+## Enhanced features
+
+- High DPI improvements.
+TODO: @dreddy-work to provide text.
+- Windows 11 style default tooltip behavior makes the tooltip remain open when mouse hovers over it, and not disappear automatically. The tooltip can be dismissed by CONTROL or ESCAPE keys.
+TODO: @mmcgaw to provide text.
```
[the remark this answers is not in the document]
Comment (merriemcgaw) [reply dotnet-desktop-guide/net/winforms/whats-new/index.md]: I don't think I'd change it or add anything. @Tanya-Solyanik?

File: dotnet-desktop-guide/net/winforms/whats-new/net50.md [hunk 6]
```diff
@@ -1,15 +1,15 @@
...
-title: What's new in Windows Forms
-description: Learn about what's new in Windows Forms for .NET. Windows Forms. .NET provides new features and enhancements over .NET Framework.
+title: What's new in Windows Forms .NET 5
+description: Learn about what's new in Windows Forms for .NET 5. Windows Forms. .NET provides new features and enhancements over .NET Framework.
 ms.date: 11/05/2020
 ms.topic: conceptual
 ---
 
-# What's new (Windows Forms .NET)
+# What's new for .NET 5 (Windows Forms .NET)
```
Comment (merriemcgaw) [inline dotnet-desktop-guide/net/winforms/whats-new/net50.md:8]: Why is this .NET 5 where everything else is .NET 6? Am I missing the purpose of the file?

Comment (merriemcgaw) [reply dotnet-desktop-guide/net/winforms/whats-new/net50.md:8]: Never mind 😊 I see the file name now. You can disregard.

File: dotnet-desktop-guide/net/winforms/whats-new/net60.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+## New application bootstrap
+
+TODO
+
+## Updated templates for C#
+
+In line with [related changes in .NET workloads](../../sdk/6.0/csharp-template-code.md), Windows Forms templates for C# have been updated to support `global using` directives, file-scoped namespaces, and nullable reference types. Because a typical Windows Forms app consist of multiple types split across multiple files, for example, Form1.cs and Form1.Designer.cs, top-level statements are notably absent from the Windows Forms templates. However, the updated templates do include application bootstrap code. This can cause incompatibility if you target an earlier version of .NET.
+
+TODO: note about .NET Framework templates. @OliaG to provide text.
```
Comment (merriemcgaw) [inline dotnet-desktop-guide/net/winforms/whats-new/net60.md:22]: As I mentioned - I think that we don't have to worry about that, but I'll confirm when installing on a clean VM.

File: dotnet-desktop-guide/net/winforms/whats-new/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dreddy-work) [reply dotnet-desktop-guide/net/winforms/whats-new/index.md]: We are still working though High Dpi issues forPermonV2 mode but it is still far from complete. In this release we made few improvements improvements around
1. Scaling ContainerControls and MDI child windows in PermonitorV2 mode. https://github.com/dotnet/winforms/pull/5032
2. Fixing issues around creating Winforms controls in the right DPI awarenes as the application.https://github.com/dotnet/winforms/pull/4796

File: dotnet-desktop-guide/net/winforms/whats-new/net60.md (hunk 7 above)
Comment (merriemcgaw) [reply dotnet-desktop-guide/net/winforms/whats-new/net60.md:22]: @RussKie I've confirmed - you can remove the ToDo about the .NET Framework templates. It's regressed in that they are optional components which are unchecked in the installer but they're still installing somehow on my clean VM.

File: dotnet-desktop-guide/net/winforms/whats-new/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (RussKie) [reply dotnet-desktop-guide/net/winforms/whats-new/index.md]: @dreddy-work for the blog I worded this section as follows. Do you have any objections?

> _We've been working through the high DPI space with the aim to get Windows Forms applications to correctly support [PerMonitorV2 mode](https://docs.microsoft.com/windows/win32/hidpi/high-dpi-desktop-application-development-on-windows#per-monitor-and-per-monitor-v2-dpi-awareness) out of the box. It is a challenging undertaking, and sadly we couldn't achieve as much as we'd hoped. Still in this release we made some progress, and we now can:_
> * _Create controls in the same DPI awarenes as the application_
> * _Correctly scale `ContainerControl`s and MDI child windows in PerMonitorV2 mode_
> scaling

@adegeo feel free to reword this the best way you see fit for this doc.

Thank you

File: dotnet-desktop-guide/net/winforms/whats-new/net60.md [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+```csharp
+Application.EnableVisualStyles();
+Application.SetCompatibleTextRenderingDefault(false);
+Application.SetDefaultFont(new Font(...));
+Application.SetHighDpiMode(HighDpiMode.SystemAware);
+```
+The parameters of these calls are configurable via [MSBuild properties](https://docs.microsoft.com/dotnet/core/project-sdk/msbuild-props-desktop#windows-forms-settings) in csproj or props files.
+The Windows Forms designer in Visual Studio 2022 is also aware of these properties (for now it only reads the default font), and can show you your application as it would look at runtime:
+
+<TODO: image goes here>
```
Comment (RussKie) [inline dotnet-desktop-guide/net/winforms/whats-new/net60.md:62]: @adegeo this section's wording is a compilation from my blog and and @richlander's blog (yet to be published). Feel free to reword as you see necessary.
I added an image to this commit that's being used in the blogs (appbootstrap.png). I don't know whether we want to add it to the docs.

Comment (RussKie) [reply dotnet-desktop-guide/net/winforms/whats-new/net60.md:62]: https://devblogs.microsoft.com/dotnet/announcing-net-6/

Comment (RussKie) [reply dotnet-desktop-guide/net/winforms/whats-new/net60.md:62]: https://devblogs.microsoft.com/dotnet/whats-new-for-visual-basic-in-visual-studio-2022/

Comment (adegeo) [reply dotnet-desktop-guide/net/winforms/whats-new/net60.md:62]: @RussKie Can you clarify something? I'm working on this section and the way this is worded makes it sound like the default font change that was previously introduced was reverted and now set via the `SetDefaultFont` method.  So if someone removes the call to that method, the default font remains the same as it was .NET Framework.

Correct?

Comment (merriemcgaw) [reply dotnet-desktop-guide/net/winforms/whats-new/net60.md:62]: > @RussKie Igor Velikorossov FTE Can you clarify something? I'm working on this section and the way this is worded makes it sound like the default font change that was previously introduced was reverted and now set via the SetDefaultFont method. So if someone removes the call to that method, the default font remains the same as it was .NET Framework.
>
>Correct?

Actually no, not quite correct. The Default Font changed early on, but lots of developers who were porting their applications from Framework, where they were laying out based on the old font found that with the new default font it looked ugly. Either they had to change the font on every form in their application or realign their layout. As a result we decided to ADD the new Application wide default font so that if a developer wanted to use the old defualt they could explicitly state that's what they wanted application wide. Make sense?

File: dotnet-desktop-guide/net/winforms/whats-new/net60.md [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+  - <xref:System.Windows.Forms.LinkClickedEventArgs.LinkStart?displayProperty=fullName>
+
+- <xref:System.Windows.Forms.NotifyIcon.Text?displayProperty=fullName> is now limited to 127 characters (from 63).
+
+## Enhanced features
+
+- High DPI improvements.
+TODO: @dreddy-work to provide text.
+- Windows 11 style default tooltip behavior makes the tooltip remain open when mouse hovers over it, and not disappear automatically. The tooltip can be dismissed by CONTROL or ESCAPE keys.
+- Microsoft UI Automation patterns work better with accessibility tools like Narrator and Jaws.
```
Comment (merriemcgaw) [inline dotnet-desktop-guide/net/winforms/whats-new/net60.md:121]: @RussKie you might rephrase this to "assistive technology tools like Narrator and Jaws" - Assistive tech is sort of the industry standard.

**Review by merriemcgaw — APPROVED** [review]

File: dotnet-desktop-guide/net/winforms/whats-new/net60.md [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+  ```
+
+## Visual Studio designer improvements
+
+TODO: Text
+TODO: Image comparing differences
+
+## New API
+
+- <xref:System.Windows.Forms.Application.SetDefaultFont?displayProperty=fullName>
```
Comment (RussKie) [inline dotnet-desktop-guide/net/winforms/whats-new/net60.md:110]: For some reason this isn't rendering correctly
image

File: dotnet-desktop-guide/net/winforms/whats-new/net60.md [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+
+  - <xref:System.Windows.Forms.LinkClickedEventArgs.LinkLength?displayProperty=fullName>
+  - <xref:System.Windows.Forms.LinkClickedEventArgs.LinkStart?displayProperty=fullName>
+
+- <xref:System.Windows.Forms.NotifyIcon.Text?displayProperty=fullName> is now limited to 127 characters (from 63).
+
+## Enhanced features
+
+- High DPI improvements.
+TODO: @dreddy-work to provide text.
```
Comment (RussKie) [inline dotnet-desktop-guide/net/winforms/whats-new/net60.md:159]: The text is here: https://github.com/dotnet/docs-desktop/pull/1199#discussion_r744406765

**Review by RussKie — COMMENTED** [review]
👍

File: dotnet-desktop-guide/net/winforms/whats-new/net60.md (hunk 9 above)
Comment (adegeo) [reply dotnet-desktop-guide/net/winforms/whats-new/net60.md:161]: @merriemcgaw does it "work better" than the last time? Because we pointed this same thing out with 5.0

File: dotnet-desktop-guide/net/winforms/whats-new/net60.md [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+|-------------------------------------------------------------------------------------------------------------------------------|-- |----------------------------------------------------------------------------------------|
+| [ApplicationVisualStyles](/dotnet/core/project-sdk/msbuild-props-desktop#applicationvisualstyles)                             | `true` | `Application.EnableVisualStyles();`                                                    |
+| [ApplicationUseCompatibleTextRendering](/dotnet/core/project-sdk/msbuild-props-desktop#applicationusecompatibletextrendering) | `false` | `Application.SetCompatibleTextRenderingDefault(false);`                                |
+| [ApplicationHighDpiMode](/dotnet/core/project-sdk/msbuild-props-desktop#applicationhighdpimode)                               | `SystemAware` | `Application.SetHighDpiMode(HighDpiMode.SystemAware);`                                 |
+| [ApplicationDefaultFont](/dotnet/core/project-sdk/msbuild-props-desktop#applicationdefaultfont)                               | `Segoe UI, 9pt` | `Application.SetDefaultFont(new Font(new FontFamily("Microsoft Sans Serif"), 8.25f));` |
```
Comment (RussKie) [inline dotnet-desktop-guide/net/winforms/whats-new/net60.md:75]: This looks strange:
image

Should this be:
```suggestion
| [ApplicationDefaultFont](/dotnet/core/project-sdk/msbuild-props-desktop#applicationdefaultfont)                               | `Microsoft Sans Serif, 8.25f` | `Application.SetDefaultFont(new Font(new FontFamily("Microsoft Sans Serif"), 8.25f));` |
```

Ditto below:
```
<ApplicationDefaultFont>Microsoft Sans Serif, 8.25f</ApplicationDefaultFont>
```

**Review by RussKie — COMMENTED** [review]
👍 with the comments

File: dotnet-desktop-guide/net/winforms/whats-new/net60.md (hunk 9 above)
Comment (RussKie) [reply dotnet-desktop-guide/net/winforms/whats-new/net60.md]: Works betterer than ".NET 5.0 better" :D

File: dotnet-desktop-guide/net/winforms/whats-new/net60.md (hunk 12 above)
Comment (adegeo) [reply dotnet-desktop-guide/net/winforms/whats-new/net60.md:75]: I think this should actually be
```suggestion
| [ApplicationDefaultFont](/dotnet/core/project-sdk/msbuild-props-desktop#applicationdefaultfont)                               | `Segoe UI, 9pt` | `Application.SetDefaultFont(new Font(new FontFamily("Segoe UI"), 9f));` |
```

Comment (adegeo) [inline dotnet-desktop-guide/net/winforms/whats-new/net60.md:75]:
```suggestion
| Project setting                                                                                                               | Default value   | Corresponding API                                                                                    |
|-------------------------------------------------------------------------------------------------------------------------------|-----------------|----------------------------------------------------------------------------------------|
| [ApplicationVisualStyles](/dotnet/core/project-sdk/msbuild-props-desktop#applicationvisualstyles)                             | `true`          | `Application.EnableVisualStyles`                                                    |
| [ApplicationUseCompatibleTextRendering](/dotnet/core/project-sdk/msbuild-props-desktop#applicationusecompatibletextrendering) | `false`         | `Application.SetCompatibleTextRenderingDefault`                                |
| [ApplicationHighDpiMode](/dotnet/core/project-sdk/msbuild-props-desktop#applicationhighdpimode)                               | `SystemAware`   | `Application.SetHighDpiMode`                                 |
| [ApplicationDefaultFont](/dotnet/core/project-sdk/msbuild-props-desktop#applicationdefaultfont)                               | `Segoe UI, 9pt` | `Application.SetDefaultFont` |
```

File: dotnet-desktop-guide/net/winforms/whats-new/net60.md [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+    <OutputType>WinExe</OutputType>
+    <TargetFramework>net6.0-windows</TargetFramework>
+    <Nullable>enable</Nullable>
+    <UseWindowsForms>true</UseWindowsForms>
+    <ImplicitUsings>enable</ImplicitUsings>
+
+    <ApplicationVisualStyles>true</ApplicationVisualStyles>
+    <ApplicationUseCompatibleTextRendering>false</ApplicationUseCompatibleTextRendering>
+    <ApplicationHighDpiMode>SystemAware</ApplicationHighDpiMode>
+    <ApplicationDefaultFont>Segoe UI, 9pt</ApplicationDefaultFont>
```
Comment (adegeo) [inline dotnet-desktop-guide/net/winforms/whats-new/net60.md:92]:
```suggestion
    <ApplicationDefaultFont>Microsoft Sans Serif, 8.25pt</ApplicationDefaultFont>
```

**Review by RussKie — COMMENTED** [review]
Nice!

**Comment by RussKie** [comment]
I can't approve my own PR :)
Feel free to merge it when you're happy

File: dotnet-desktop-guide/net/winforms/whats-new/net60.md [hunk 14]
```diff
@@ -99,31 +142,79 @@ The default font can now be set in two ways:
...
+- <xref:System.Windows.Forms.Design.TreeViewDesigner?displayProperty=fullName>
+- <xref:System.Windows.Forms.Design.UpDownBaseDesigner?displayProperty=fullName>
+- <xref:System.Windows.Forms.Design.UserControlDocumentDesigner?displayProperty=fullName>
+
+## High DPI improvements for PerMonitorV2
+
+High DPI rendering with <xref:System.Windows.Forms.HighDpiMode.PerMonitorV2> have been improved:
+
+- Controls are created with the same DPI awareness as the application.
+- Container controls and MDI child windows now scale correctly.
```
Comment (dreddy-work) [inline dotnet-desktop-guide/net/winforms/whats-new/net60.md:190]:
```suggestion
- Container controls and MDI child windows have improved scaling behaviors.
```

**Review by adegeo — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you!

**Comment by adegeo** [comment]
Fixes #1191
